nir/ir.py

```diff
@@ -30,9 +30,39 @@ class NIRGraph(NIRNode):
     def from_list(*nodes: NIRNode) -> "NIRGraph":
         """Create a sequential graph from a list of nodes by labelling them after
         indices."""
+
+        def unique_node_name(node, name_dict):
+            print(node)
+            basename = node.__class__.__name__.lower()
+            print(basename)
+            idx = None
+            if basename in name_dict.keys():
```

Comment:
Another way to do this would be to have a defaultdict

```python
name_dict = defaultdict(int)
id = name_dict[basename]
name_dict[basename] += 1
node_name = f"{basename}" + f"_{id}" if id > 0 else ""
```

Or a [`Counter`](https://docs.python.org/3/library/collections.html#collections.Counter)
```python
appearances = Counter()
count = appearances[basename]
node_name = f"{basename}" + "_{count} if count > 0 else ""
appearances.update([basename])
```